packages/cli/config/eslint-factory.js

```diff
@@ -129,6 +129,17 @@ function createConfig(dir, extraConfig = {}) {
           paths: [
             ...(restrictedImports ?? []),
             ...(restrictedSrcImports ?? []),
+            {
+              name: '@material-ui/core',
+              importNames: ['Link'],
+              message:
+                'Prefer using `Link` from `@backstage/core-components` rather than material-UI',
+            },
+            {
+              name: '@material-ui/core/Link',
+              message:
+                'Prefer using `Link` from `@backstage/core-components` rather than material-UI',
+            },
```

Comment:
Hmm, bit worried about shipping this without any way to disable it. It's a breaking change after all and one that might not be trivial to fix in many projects. I'm also not sure it makes sense to enforce usage globally, it could be enough to require it in this repo.

Either way, I think we want to move this down to where we restrict `@material-ui/icons` further down. Let's start there